src/components/Requirements.vue

```diff
@@ -1,214 +1,573 @@
+            singleMenuRequirement.fulfilled = req.fulfilled;
+            singleMenuRequirement.required = req.required;
+          }
+
+          // Default display value of false for all requirement lists
+          req.display = false;
+
+          if (!req.fulfilled || req.fulfilled < req.required) {
+            singleMenuRequirement.ongoing.push(req);
+          } else {
+            singleMenuRequirement.completed.push(req);
+          }
+        });
+
+        if (!singleMenuRequirement.type) {
+          singleMenuRequirement.type = 'Requirements';
+          singleMenuRequirement.fulfilled = singleMenuRequirement.completed.length;
+          singleMenuRequirement.required = singleMenuRequirement.ongoing.length + singleMenuRequirement.completed.length;
+        }
+
+        this.reqs.push(singleMenuRequirement);
+      });
+    });
+  },
+
   data() {
     return {
+      reqsData,
+      actives: [false],
+      modalShow: false,
       reqs: [
-        {
-          name: 'College Requirements',
-          type: 'College',
-          count: 'Credits',
-          progress: 46,
-          total: 120,
-          color: '#2BBCC6',
-          displayDetails: false,
-          ongoing: [
-            {
-              name: 'CALS Credits',
-              count: 'Credits',
-              progress: 12,
-              total: 55
-            },
-            {
-              name: 'PE Credits',
-              count: 'Credits',
-              progress: 1,
-              total: 2
-            }
-          ],
-          displayCompleted: true,
-          completed: [
-            {
-              name: 'Quantitative Literacy',
-              count: 'Credits',
-              progress: 2,
-              total: 2
-            },
-            {
-              name: 'Chemistry/Physics',
-              count: 'Credits',
-              progress: 1,
-              total: 2
-            }
-          ]
-        },
-        {
-          name: 'Major Requirements',
-          type: 'Major',
-          count: 'Courses',
-          progress: 4,
-          total: 8,
-          color: '#4D53DC',
-          displayDetails: false,
-          ongoing: [
-            {
-              name: 'CALS Credits',
-              count: 'Credits',
-              progress: 12,
-              total: 55
-            },
-            {
-              name: 'PE Credits',
-              count: 'Credits',
-              progress: 1,
-              total: 2
-            }
-          ],
-          displayCompleted: true,
-          completed: [
-            {
-              name: 'Quantitative Literacy',
-              count: 'Credits',
-              progress: 2,
-              total: 2
-            },
-            {
-              name: 'Chemistry/Physics',
-              count: 'Credits',
-              progress: 1,
-              total: 2
-            }
-          ]
-        }
+        // Data structure for menu
+        // {
+        //   name: 'UNIVERSITY REQUIREMENT',
```

Comment:
Commented out code?

Reply:
Mostly used to remind me of the data structure that the menu needs in order to work properly. The menu still needs testing and additional features (such as drag and drop recommended courses), so I'm leaving this in there for now